src/message.ts

```diff
@@ -18,16 +25,24 @@ export abstract class Message {
     return this.jobRef
   }
 
+  getId(): number {
+    return this.id
+  }
+
   hasJobRef(): boolean {
     return !this.jobRef.isNull()
   }
 
+  hasId(): boolean {
+    return this.id !== NO_JOB_ID
```

Comment:
Remember to replace by `this.jobId.isNull()` although I think we were discussing that right now there is not any case where a message does not have a Job ID, right?
In that case, I would not create a null object for JobId and I would remove this method because messages always have a JobId.